lib/matplotlib/rcsetup.py

```diff
@@ -310,6 +318,25 @@ def validate_colorlist(s):
         msg = "'s' must be of type [ string | list | tuple ]"
         raise ValueError(msg)
 
+def validate_cycler(s):
+    'return a cycler object from a string repr or the object itself'
+    if isinstance(s, six.string_types):
+        try:
+            # TODO: We might want to rethink this...
+            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
```

Comment:
It would look exactly like how you would create cycler objects in code,
just with quotes around it. It is so we can specify them in the rc file.
I'll be adding unit tests for this later today as well as documentation.
On Aug 15, 2015 12:51 AM, "OceanWolf" <notifications@github.com> wrote:

> In lib/matplotlib/rcsetup.py
> <https://github.com/matplotlib/matplotlib/pull/4686#discussion_r37133363>:
>
> > @@ -310,6 +318,25 @@ def validate_colorlist(s):
> >          msg = "'s' must be of type [ string | list | tuple ]"
> >          raise ValueError(msg)
> >
> > +def validate_cycler(s):
> > +    'return a cycler object from a string repr or the object itself'
> > +    if isinstance(s, six.string_types):
> > +        try:
> > +            # TODO: We might want to rethink this...
> > +            cycler_inst = eval(s, {'cycler': cycler, 'Cycler': Cycler})
>
> Can you give an example of passing a string to this function?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4686/files#r37133363>.
>